src/pages/saksbehandling/revurdering/opplysningsplikt/Opplysningsplikt.tsx

```diff
@@ -0,0 +1,208 @@
+import { Button, Panel, Select } from '@navikt/ds-react';
+import React from 'react';
+import { Controller, useFieldArray, useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+
+import ApiErrorAlert from '~src/components/apiErrorAlert/ApiErrorAlert';
+import DatePicker from '~src/components/datePicker/DatePicker';
+import ToKolonner from '~src/components/toKolonner/ToKolonner';
+import { lagreOpplysningsplikt } from '~src/features/revurdering/revurderingActions';
+import { useAsyncActionCreator } from '~src/lib/hooks';
+import { useI18n } from '~src/lib/i18n';
+import { getDateErrorMessage } from '~src/lib/validering';
+import { RevurderingBunnknapper } from '~src/pages/saksbehandling/revurdering/bunnknapper/RevurderingBunnknapper';
+import * as sharedStyles from '~src/pages/saksbehandling/revurdering/revurdering.module.less';
+import RevurderingsperiodeHeader from '~src/pages/saksbehandling/revurdering/revurderingsperiodeheader/RevurderingsperiodeHeader';
+import { OpplysningspliktBeksrivelse } from '~src/types/grunnlagsdataOgVilkårsvurderinger/opplysningsplikt/Opplysningsplikt';
+import { RevurderingStegProps } from '~src/types/Revurdering';
+import { parseIsoDateOnly, sluttenAvMåneden, toIsoDateOnlyString } from '~src/utils/date/dateUtils';
+
+import UtfallSomIkkeStøttes from '../utfallSomIkkeStøttes/UtfallSomIkkeStøttes';
+
+import GjeldendeOpplysningsplikt from './GjeldendeOpplysningsplikt';
+import messages from './opplysningsplikt-nb';
+import * as styles from './opplysningsplikt.module.less';
+import { OpplysningspliktVilkårForm, schemaValidation } from './OpplysningspliktUtils';
+
+const Opplysningsplikt = (props: RevurderingStegProps) => {
+    const navigate = useNavigate();
+    const { formatMessage } = useI18n({ messages: { ...messages } });
+    const [status, lagre] = useAsyncActionCreator(lagreOpplysningsplikt);
+
+    const vurderinger = props.revurdering.grunnlagsdataOgVilkårsvurderinger.opplysningsplikt?.vurderinger ?? [
+        { periode: props.revurdering.periode, beskrivelse: undefined },
+    ];
+
+    const revurderingsperiode = {
+        fraOgMed: new Date(props.revurdering.periode.fraOgMed),
+        tilOgMed: new Date(props.revurdering.periode.tilOgMed),
+    };
+
+    const form = useForm<OpplysningspliktVilkårForm>({
+        resolver: yupResolver(schemaValidation),
+        defaultValues: {
+            opplysningsplikt: vurderinger.map((vurdering) => ({
+                periode: {
+                    fraOgMed: parseIsoDateOnly(vurdering.periode.fraOgMed),
+                    tilOgMed: parseIsoDateOnly(vurdering.periode.tilOgMed),
+                },
+                beskrivelse: vurdering.beskrivelse ?? null,
+            })),
+        },
+    });
+
+    const { fields, append, remove } = useFieldArray({
+        name: 'opplysningsplikt',
+        control: form.control,
+    });
+
+    const handleSubmit = async (form: OpplysningspliktVilkårForm, gåtil: 'neste' | 'avbryt') => {
```

Comment:
Hvorfor ikke bara ta in urlen direkte, d.v.s å sende in props.neste eller props.tilbake så slipper man ha logikk i callbacken